qa-test

```diff
@@ -2,37 +2,81 @@
 
 set -ex
 
-##################
-### Slow tests ###
-##################
+test_relayd()
+{
+  cd relay/sources/relayd && make check
+}
+
+test_language()
+{
+  cd rudder-lang && make check
+}
+
+test_webapp_mvn()
+{
+  cd webapp/sources && mvn clean install -Dmaven.test.postgres=false
+}
+
+test_webapp_hooks()
+{
+  webapp/sources/rudder/rudder-core/src/test/resources/hooks.d/test-hooks.sh
+}
+
+test_rudder_pkg()
+{
+  cd relay/sources/ && make check && cd -
+}
+
+test_shell()
+{
+  mkdir -p .shellcheck
+  find . \( -path ./.git -prune -o -path ./rudder-lang/target -prune -o -path ./relay/sources/relayd/target -prune \) -o -type f -exec grep -Eq '^#!(.*/|.*env +)(sh|bash|ksh)' {} \; -print |
+    while IFS="" read -r file
+    do
+      shellcheck --format=checkstyle "$file" > .shellcheck/$(basename ${file}).log || true
+    done
+}
+
+# fails on error and ignores other levels
+test_shell_error()
+{
+  # Shellcheck
+  find . \( -path ./.git -prune -o -path ./rudder-lang/target -prune -o -path ./relay/sources/relayd/target -prune \) -o -type f -exec grep -Eq '^#!(.*/|.*env +)(sh|bash|ksh)' {} \; -print |
+    while IFS="" read -r file
+    do
+      # with recent shellcheck, "-S error" replaces this hack
+      shellcheck -f gcc "$file" | grep " error: " && exit 1 || true
+    done
+}
+
+# fails on error and ignores other levels
+test_python_error()
+{
+  find . ! -wholename '*rudder-lang/repos/*' ! -name 'convertOpenLDAPSchema.py' ! -name 'systemctl3.py' ! -wholename '*jsondiff/*' -name '*.py' | xargs pylint -E --persistent=n --disable=C,R,import-error,no-member,no-name-in-module
+}
+
+#####
+
 if [ "$1" = "--relayd" ]; then
-  cd relay/sources && make check
+  test_relayd
   exit 0
 elif [ "$1" = "--language" ]; then
-  cd rudder-lang && make test
+  test_language
+  exit 0
+elif [ "$1" = "--rudder-pkg" ]; then
+  test_rudder_pkg
   exit 0
 elif [ "$1" = "--scala" ]; then
-  cd webapp/sources && mvn clean install -Dmaven.test.postgres=false
+  test_webapp_hooks
+  test_webapp_mvn
+  exit 0
+elif [ "$1" = "--shell" ]; then
+  test_shell
   exit 0
+else
```

Comment:
merge already passes --merge as an argument

Reply:
do we want a different behavior when no arguments are passed?

Reply:
i would have said run the longest tests